Unable to capitalize

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I am unable to capitalize my characters unless I hold down the caps lock. I checked with other applications which was fine, only in Zed I need to hold down my caps lock. Before recent official release of Zed on Linux, I was using the one where you had to build it yourself, that one has same the issue too.
This is on Linux system Ubuntu 24.04 LTS.

### Environment

Zed: v0.143.6 (Zed)
OS: Linux X11 ubuntu 24.04
Memory: 62.5 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:31.121Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:31.121Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:31-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:31.278Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:31.278Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:31-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:31.382Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 104 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:31.382Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 104 ms"
  ]
}
2024-07-11T19:10:31-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:31.382Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:31.382Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:10:31-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:31.382Z] request done: headerRequestId: [427e783f-45ba-4ef2-b20e-99c55a2b303c] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:31.382Z]",
  "extra": [
    "request done: headerRequestId: [427e783f-45ba-4ef2-b20e-99c55a2b303c] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:10:31-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:31.571Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:31.571Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:31-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:31.763Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:31.763Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:31-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:31.938Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:31.938Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:32-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:32.105Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:32.105Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:32-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:32.273Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:32.273Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:32-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:32.464Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:32.464Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:32-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:32.577Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 113 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:32.577Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 113 ms"
  ]
}
2024-07-11T19:10:32-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:32.578Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:32.578Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:10:32-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:32.578Z] request done: headerRequestId: [a6fe0606-a0e1-4691-9fc4-eb731b11715d] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:32.578Z]",
  "extra": [
    "request done: headerRequestId: [a6fe0606-a0e1-4691-9fc4-eb731b11715d] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:10:32-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:10:32-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:32.728Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:32.728Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:10:33-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:10:33-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:33.582Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:33.582Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:33-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:33.683Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 101 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:33.683Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 101 ms"
  ]
}
2024-07-11T19:10:33-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:33.684Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:33.684Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:10:33-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:33.684Z] request done: headerRequestId: [905f0027-f5ac-4727-b358-4930b11566df] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:33.684Z]",
  "extra": [
    "request done: headerRequestId: [905f0027-f5ac-4727-b358-4930b11566df] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:10:33-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:33.776Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:33.776Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:10:34-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:34.051Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:34.051Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:34-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:34.195Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 144 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:34.195Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 144 ms"
  ]
}
2024-07-11T19:10:34-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:34.196Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:34.196Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:10:34-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:34.196Z] request done: headerRequestId: [97007b90-f842-46dd-8f24-74537ac6a833] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:34.196Z]",
  "extra": [
    "request done: headerRequestId: [97007b90-f842-46dd-8f24-74537ac6a833] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:10:35-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:35.028Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:35.028Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:10:35-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:35.128Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:35.128Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:35-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:35.369Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:35.369Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:35-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:35.684Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:35.684Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:35-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:35.928Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 244 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:35.928Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 244 ms"
  ]
}
2024-07-11T19:10:35-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:35.928Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:35.928Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:10:35-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:10:35.928Z] request done: headerRequestId: [25deca05-30fc-4a1d-b2f5-ef84e3083b63] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:10:35.928Z]",
  "extra": [
    "request done: headerRequestId: [25deca05-30fc-4a1d-b2f5-ef84e3083b63] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:10:36-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:36.346Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:36.346Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:36-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:36.558Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:36.558Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:10:36-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:36.712Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:36.712Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:36-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:36.877Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:36.877Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:37-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:37.093Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:37.093Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:37-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:37.271Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:37.271Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:10:37-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:37.413Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:37.413Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:10:37-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:37.484Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:37.484Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:10:37-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:37.689Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:37.689Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:37-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:10:37.861Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:10:37.861Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:10:38-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:38.004Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:38.004Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:10:38-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:38.347Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:38.347Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:10:38-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:38.574Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:38.574Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:10:38-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:38.749Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:38.749Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:10:41-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:10:45-07:00 [INFO] open paths ["/home/xuan/.config/zed/settings.json"]
2024-07-11T19:10:45-07:00 [INFO] Initializing default prettier with plugins {}
2024-07-11T19:10:45-07:00 [INFO] starting language servers for JSONC: json-language-server
2024-07-11T19:10:45-07:00 [INFO] starting language server "json-language-server", path: "/home/xuan/.config/zed", id: 2
2024-07-11T19:10:45-07:00 [INFO] Initializing default prettier with plugins {}
2024-07-11T19:10:45-07:00 [INFO] Initializing default prettier with plugins {}
2024-07-11T19:10:45-07:00 [INFO] starting language servers for Python: pyright
2024-07-11T19:10:45-07:00 [INFO] starting language servers for JSONC: json-language-server
2024-07-11T19:10:45-07:00 [INFO] fetching latest version of language server "json-language-server"
2024-07-11T19:10:45-07:00 [INFO] Node runtime install_if_needed
2024-07-11T19:10:45-07:00 [INFO] language server 1: unregistered workspace/DidChangeWatchedFiles capability with id 636bff29-4104-4927-95e1-4667eabb8677
2024-07-11T19:10:45-07:00 [INFO] Initializing default prettier with plugins {}
2024-07-11T19:10:45-07:00 [INFO] Initializing default prettier with plugins {}
2024-07-11T19:10:45-07:00 [INFO] starting language servers for Python: pyright
2024-07-11T19:10:45-07:00 [INFO] starting language servers for JSONC: json-language-server
2024-07-11T19:10:45-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:45.371Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:45.371Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:10:45-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:10:45.463Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:10:45.463Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:10:45-07:00 [INFO] language server 1: unregistered workspace/DidChangeWatchedFiles capability with id eeb64d55-dba6-47ff-9617-25c999841809
2024-07-11T19:10:45-07:00 [INFO] downloading language server "json-language-server"
2024-07-11T19:10:45-07:00 [INFO] Node runtime install_if_needed
2024-07-11T19:10:45-07:00 [INFO] starting language server. binary path: "/home/xuan/.local/share/zed/node/node-v18.15.0-linux-x64/bin/node", working directory: "/home/xuan/.config/zed", args: ["/home/xuan/.local/share/zed/languages/json-language-server/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server", "--stdio"]
2024-07-11T19:10:48-07:00 [INFO] open paths ["/home/xuan/.config/zed/settings.json"]
2024-07-11T19:10:53-07:00 [INFO] open paths ["/home/xuan/.config/zed/keymap.json"]
2024-07-11T19:10:53-07:00 [INFO] Initializing default prettier with plugins {}
2024-07-11T19:10:53-07:00 [INFO] starting language servers for JSONC: json-language-server
2024-07-11T19:10:56-07:00 [INFO] open paths ["/home/xuan/.config/zed/settings.json"]
2024-07-11T19:11:00-07:00 [ERROR] oneshot canceled
2024-07-11T19:11:00-07:00 [INFO] Initializing default prettier with plugins {}
2024-07-11T19:11:00-07:00 [INFO] starting language servers for Python: pyright
2024-07-11T19:11:00-07:00 [INFO] language server 1: unregistered workspace/DidChangeWatchedFiles capability with id d7cc6e66-aef0-4e24-92f8-d66d7004527a
2024-07-11T19:11:00-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:00.536Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:00.536Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:00-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:00.626Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 90 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:00.626Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 90 ms"
  ]
}
2024-07-11T19:11:00-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:11:00.627Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:11:00.627Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:11:00-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:11:00.627Z] request done: headerRequestId: [81a639d0-e3bd-4965-a270-ba44868f7e15] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:11:00.627Z]",
  "extra": [
    "request done: headerRequestId: [81a639d0-e3bd-4965-a270-ba44868f7e15] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:11:01-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 211 }
2024-07-11T19:11:47-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:11:47-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:47.508Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:47.508Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:47-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:47.784Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:47.784Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:48-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:48.070Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:48.070Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:48-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:48.270Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:48.270Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:11:48-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:11:48-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:48.408Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:48.408Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:11:48-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:11:48-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:48.552Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:48.552Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:11:48-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:48.703Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:48.703Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:48-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:48.801Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 98 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:48.801Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 98 ms"
  ]
}
2024-07-11T19:11:48-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:11:48.801Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:11:48.801Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:11:48-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:11:48.801Z] request done: headerRequestId: [276f40be-580e-42bc-8700-018206e8894c] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:11:48.801Z]",
  "extra": [
    "request done: headerRequestId: [276f40be-580e-42bc-8700-018206e8894c] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:11:49-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:49.573Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:49.573Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:11:50-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:50.824Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:50.824Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:11:50-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:50.969Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:50.969Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:51-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:51.125Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 156 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:51.125Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 156 ms"
  ]
}
2024-07-11T19:11:51-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:11:51.125Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:11:51.125Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:11:51-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:11:51.125Z] request done: headerRequestId: [36599168-2e6b-4921-bff1-47b9d6116312] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:11:51.125Z]",
  "extra": [
    "request done: headerRequestId: [36599168-2e6b-4921-bff1-47b9d6116312] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:11:51-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:51.445Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:51.445Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:51-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:51.615Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:51.615Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:11:51-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:51.754Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:51.754Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:11:51-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:51.908Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:51.908Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:52.064Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:52.064Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:11:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:11:52.215Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:11:52.215Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:11:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:11:52.316Z] Found inline suggestions locally",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:11:52.316Z]",
  "extra": [
    "Found inline suggestions locally"
  ]
}
2024-07-11T19:11:52-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:12:12-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 213 }
2024-07-11T19:12:52-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:12:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:52.293Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:52.293Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:52.681Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:52.681Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:52.938Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 256 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:52.938Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 256 ms"
  ]
}
2024-07-11T19:12:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:12:52.940Z] solution 2 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:12:52.940Z]",
  "extra": [
    "solution 2 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:12:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:12:52.941Z] solution 1 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:12:52.941Z]",
  "extra": [
    "solution 1 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:12:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:12:52.945Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:12:52.945Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:12:52-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:12:52.945Z] request done: headerRequestId: [95737473-a800-4a0d-b06b-84c52a80c480] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:12:52.945Z]",
  "extra": [
    "request done: headerRequestId: [95737473-a800-4a0d-b06b-84c52a80c480] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:12:53-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:53.415Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:53.415Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:53-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:12:53-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:53.586Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:53.586Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:53-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:12:53-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:53.795Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:53.795Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:53-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:12:53.985Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:12:53.985Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:12:54-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:54.126Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:54.126Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:54-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:54.297Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:54.297Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:54-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:12:54-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:54.481Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:54.481Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:54-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:54.636Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:54.636Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:54-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:54.743Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 107 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:54.743Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 107 ms"
  ]
}
2024-07-11T19:12:54-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:12:54.744Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:12:54.744Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:12:54-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:12:54.744Z] request done: headerRequestId: [a69a78a5-2a98-4865-a7d5-0760b413769b] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:12:54.744Z]",
  "extra": [
    "request done: headerRequestId: [a69a78a5-2a98-4865-a7d5-0760b413769b] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:12:55-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:12:55-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:12:55.748Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:12:55.748Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:12:55-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:12:55.921Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:12:55.921Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [ghostText] [2024-07-12T02:12:56.037Z] Cancelled during debounce",
  "metadataStr": "[INFO] [ghostText] [2024-07-12T02:12:56.037Z]",
  "extra": [
    "Cancelled during debounce"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:56.166Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:56.166Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:56.346Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:56.346Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:56.512Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:56.512Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:56.667Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:56.667Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] ignoring macOS specific set_edited
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:56.843Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:56.843Z]",
  "extra": [
    "[fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [default] [2024-07-12T02:12:56.950Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 107 ms",
  "metadataStr": "[INFO] [default] [2024-07-12T02:12:56.950Z]",
  "extra": [
    "request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 107 ms"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:12:56.951Z] solution 0 returned. finish reason: [stop]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:12:56.951Z]",
  "extra": [
    "solution 0 returned. finish reason: [stop]"
  ]
}
2024-07-11T19:12:56-07:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 1,
  "message": "[INFO] [streamChoices] [2024-07-12T02:12:56.951Z] request done: headerRequestId: [cbedc56e-797d-4b74-9c8e-3f7e14675dec] model deployment ID: [d031-20240607173348]",
  "metadataStr": "[INFO] [streamChoices] [2024-07-12T02:12:56.951Z]",
  "extra": [
    "request done: headerRequestId: [cbedc56e-797d-4b74-9c8e-3f7e14675dec] model deployment ID: [d031-20240607173348]"
  ]
}
2024-07-11T19:12:58-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 215 }
2024-07-11T19:13:19-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 218 }
2024-07-11T19:13:21-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 220 }
2024-07-11T19:13:43-07:00 [INFO] new;
2024-07-11T19:13:43-07:00 [INFO] keep_updated;
2024-07-11T19:13:43-07:00 [INFO] new;
2024-07-11T19:13:43-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 223 }
2024-07-11T19:14:20-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 225 }
2024-07-11T19:15:19-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 226 }
2024-07-11T19:15:23-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 229 }
2024-07-11T19:15:29-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 232 }
2024-07-11T19:16:59-07:00 [INFO] open paths ["/home/xuan/.local/share/zed/logs/Zed.log"]
2024-07-11T19:17:01-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:01-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:02-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:02-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:02-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:02-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:03-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:03-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:03-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:22-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:22-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:22-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:24-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:24-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:24-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:25-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:25-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:25-07:00 [WARN] Unknown button press: ButtonPressEvent { .. }
2024-07-11T19:17:30-07:00 [WARN] Unknown request DestroyIcReply { input_method_id: 9, input_context_id: 233 }

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/gpui/src/platform/linux/x11/client.rs
+++ b/crates/gpui/src/platform/linux/x11/client.rs
@@ -24,7 +24,7 @@ use x11rb::xcb_ffi::XCBConnection;
 use xim::{x11rb::X11rbClient, Client};
 use xim::{AttributeName, InputStyle};
 use xkbc::x11::ffi::{XKB_X11_MIN_MAJOR_XKB_VERSION, XKB_X11_MIN_MINOR_XKB_VERSION};
-use xkbcommon::xkb as xkbc;
+use xkbcommon::xkb::{self as xkbc, LayoutIndex, ModMask};
 
 use crate::platform::linux::LinuxClient;
 use crate::platform::{LinuxCommon, PlatformWindow};
@@ -94,6 +94,13 @@ impl From<xim::ClientError> for EventHandlerError {
     }
 }
 
+#[derive(Debug, Default, Clone)]
+struct XKBStateNotiy {
+    depressed_layout: LayoutIndex,
+    latched_layout: LayoutIndex,
+    locked_layout: LayoutIndex,
+}
+
 pub struct X11ClientState {
     pub(crate) loop_handle: LoopHandle<'static, X11Client>,
     pub(crate) event_loop: Option<calloop::EventLoop<'static, X11Client>>,
@@ -113,6 +120,7 @@ pub struct X11ClientState {
     pub(crate) mouse_focused_window: Option<xproto::Window>,
     pub(crate) keyboard_focused_window: Option<xproto::Window>,
     pub(crate) xkb: xkbc::State,
+    previous_xkb_state: XKBStateNotiy,
     pub(crate) ximc: Option<X11rbClient<Rc<XCBConnection>>>,
     pub(crate) xim_handler: Option<XimHandler>,
     pub modifiers: Modifiers,
@@ -350,6 +358,7 @@ impl X11Client {
             mouse_focused_window: None,
             keyboard_focused_window: None,
             xkb: xkb_state,
+            previous_xkb_state: XKBStateNotiy::default(),
             ximc,
             xim_handler,
 
@@ -622,7 +631,11 @@ impl X11Client {
                     event.latched_group as u32,
                     event.locked_group.into(),
                 );
-
+                state.previous_xkb_state = XKBStateNotiy {
+                    depressed_layout: event.base_group as u32,
+                    latched_layout: event.latched_group as u32,
+                    locked_layout: event.locked_group.into(),
+                };
                 let modifiers = Modifiers::from_xkb(&state.xkb);
                 if state.modifiers == modifiers {
                     drop(state);
@@ -644,11 +657,18 @@ impl X11Client {
                 let modifiers = modifiers_from_state(event.state);
                 state.modifiers = modifiers;
                 state.pre_ime_key_down.take();
-
                 let keystroke = {
                     let code = event.detail.into();
+                    let xkb_state = state.previous_xkb_state.clone();
+                    state.xkb.update_mask(
+                        event.state.bits() as ModMask,
+                        0,
+                        0,
+                        xkb_state.depressed_layout,
+                        xkb_state.latched_layout,
+                        xkb_state.locked_layout,
+                    );
                     let mut keystroke = crate::Keystroke::from_xkb(&state.xkb, modifiers, code);
-                    state.xkb.update_key(code, xkbc::KeyDirection::Down);
                     let keysym = state.xkb.key_get_one_sym(code);
                     if keysym.is_modifier_key() {
                         return Some(());
@@ -707,8 +727,16 @@ impl X11Client {
 
                 let keystroke = {
                     let code = event.detail.into();
+                    let xkb_state = state.previous_xkb_state.clone();
+                    state.xkb.update_mask(
+                        event.state.bits() as ModMask,
+                        0,
+                        0,
+                        xkb_state.depressed_layout,
+                        xkb_state.latched_layout,
+                        xkb_state.locked_layout,
+                    );
                     let keystroke = crate::Keystroke::from_xkb(&state.xkb, modifiers, code);
-                    state.xkb.update_key(code, xkbc::KeyDirection::Up);
                     let keysym = state.xkb.key_get_one_sym(code);
                     if keysym.is_modifier_key() {
                         return Some(());

